### ISSUE TITLE
Fix for issue #3160, is not fully working.

### ISSUE THREAD

**Issue by jeanremacle** [flyway/flyway#3179]
###### Which version and edition of Flyway are you using?
7.9.1 Community

###### If this is not the latest version, can you reproduce the issue with the latest one as well? (Many bugs are fixed in newer releases and upgrading will often resolve the issue)
Yes

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)
Command-line

###### Which database are you using? (Type & version)
Oracle 19c EE

###### Which operating system are you using?
DB: Oracle Linux 7.9
Client: Windows 8.1

###### What did you do? (Please include the content causing the issue, any relevant configuration settings, the SQL statement(s) that failed (if any), and the command you ran)

I have tested the latest version 7.9.1, it is not fixed. 
The Additional regex (`ACCESSIBLE_BY_REGEX`) in the file `OracleParser.java` does not contain the package name to which we grant the access. Below is an example:

```sql
create or replace package dba_toolbox.data_core_pkg
  accessible by (package **data_pkg**)
  as
```

- When I remove the `accessible by` line, it works fine.
- Below is the terminal output:

```bash
H:\wd\GIT\dba_toolbox\database\DBA_TOOLBOX>flyway -configFiles=P:\Users\eri6d\.config\flyway.conf,H:\wd\GIT\dba_toolbox\database\DBA_TOOLBOX\flyway.conf migrate

Picked up JAVA_TOOL_OPTIONS: -Duser.language=en -Duser.region=US -Dfile.encoding=UTF-8

Flyway Community Edition 7.9.1 by Redgate

Database: jdbc:oracle:thin:@//oracle2604:1621/PDBD_UTEST (Oracle 19.0)

Successfully validated 24 migrations (execution time 00:00.468s)

Current version of schema "DBA_TOOLBOX": 0.0

ERROR: Unable to parse statement in H:\wd\GIT\dba_toolbox\database\DBA_TOOLBOX\sources\R__data_core_pkg.pks at line 1 col 1. See https://flywaydb.org/documentation/knownparserlimitations for more information: Flyway parsing bug: unable to decrease block depth below 0
Caused by: Flyway parsing bug: unable to decrease block depth below 0
```

###### What did you expect to see?

```bash
H:\wd\GIT\dba_toolbox\database\DBA_TOOLBOX>flyway -configFiles=P:\Users\eri6d\.config\flyway.conf,H:\wd\GIT\dba_toolbox\database\DBA_TOOLBOX\flyway.conf migrate
Picked up JAVA_TOOL_OPTIONS: -Duser.language=en -Duser.region=US -Dfile.encoding=UTF-8
Flyway Community Edition 7.9.0 by Redgate
Database: jdbc:oracle:thin:@//oracle2604:1621/PDBD_UTEST (Oracle 19.0)
Successfully validated 24 migrations (execution time 00:00.449s)
Current version of schema "DBA_TOOLBOX": 0.0
Migrating schema "DBA_TOOLBOX" with repeatable migration "ase pkg"
Migrating schema "DBA_TOOLBOX" with repeatable migration "ase pkg body"
Migrating schema "DBA_TOOLBOX" with repeatable migration "data core pkg"
Migrating schema "DBA_TOOLBOX" with repeatable migration "data core pkg body"
Migrating schema "DBA_TOOLBOX" with repeatable migration "data pkg"
Migrating schema "DBA_TOOLBOX" with repeatable migration "data pkg body"
Migrating schema "DBA_TOOLBOX" with repeatable migration "get current scn fct"
Migrating schema "DBA_TOOLBOX" with repeatable migration "get env fct"
Migrating schema "DBA_TOOLBOX" with repeatable migration "param pkg"
Migrating schema "DBA_TOOLBOX" with repeatable migration "param pkg body"
Migrating schema "DBA_TOOLBOX" with repeatable migration "roles pkg"
Migrating schema "DBA_TOOLBOX" with repeatable migration "roles pkg body"
Migrating schema "DBA_TOOLBOX" with repeatable migration "session pkg"
Migrating schema "DBA_TOOLBOX" with repeatable migration "session pkg body"
Migrating schema "DBA_TOOLBOX" with repeatable migration "stats pkg"
Migrating schema "DBA_TOOLBOX" with repeatable migration "stats pkg body"
Migrating schema "DBA_TOOLBOX" with repeatable migration "test users pkg"
Migrating schema "DBA_TOOLBOX" with repeatable migration "test users pkg body"
WARNING: DB: Warning: execution completed with warning (SQL State: 99999 - Error Code: 17110)
Migrating schema "DBA_TOOLBOX" with repeatable migration "trace log"
Migrating schema "DBA_TOOLBOX" with repeatable migration "trace pkg"
Migrating schema "DBA_TOOLBOX" with repeatable migration "trace pkg body"
Migrating schema "DBA_TOOLBOX" with repeatable migration "users pkg"
Migrating schema "DBA_TOOLBOX" with repeatable migration "users pkg body"
Successfully applied 23 migrations to schema "DBA_TOOLBOX" (execution time 00:02.808s)
```

###### What did you see instead?

```bash
H:\wd\GIT\dba_toolbox\database\DBA_TOOLBOX>flyway -configFiles=P:\Users\eri6d\.config\flyway.conf,H:\wd\GIT\dba_toolbox\database\DBA_TOOLBOX\flyway.conf migrate

Picked up JAVA_TOOL_OPTIONS: -Duser.language=en -Duser.region=US -Dfile.encoding=UTF-8

Flyway Community Edition 7.9.1 by Redgate

Database: jdbc:oracle:thin:@//oracle2604:1621/PDBD_UTEST (Oracle 19.0)

Successfully validated 24 migrations (execution time 00:00.468s)

Current version of schema "DBA_TOOLBOX": 0.0

ERROR: Unable to parse statement in H:\wd\GIT\dba_toolbox\database\DBA_TOOLBOX\sources\R__data_core_pkg.pks at line 1 col 1. See https://flywaydb.org/documentation/knownparserlimitations for more information: Flyway parsing bug: unable to decrease block depth below 0
Caused by: Flyway parsing bug: unable to decrease block depth below 0
```

Regards,
Jean Remacle

**Comment by juliahayward** [comment]
We're looking at this and should have a fix in the next version.

**Comment by DoodleBobBuffPants** [comment]
This should be fixed by 1ee6dd176b6fb8cf9d829fd44d8fc60314b9e179, so closing for now. Do let us know if there continue to be any problems from the next release
